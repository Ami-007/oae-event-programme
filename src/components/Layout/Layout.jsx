import "./Layout.scss";
import getFontType from "../../utils/stringHelper";

const Layout = (props) => {
  const { fontType, children } = props;
  let theme = getFontType(fontType);

  return <div className={theme}>{children}</div>;
};

export default Layout;
