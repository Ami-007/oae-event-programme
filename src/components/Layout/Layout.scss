@use "../../assets/sass/typography" as *;
@use "../../assets/sass/variables" as *;
:root {
  --theme-color-primary: #ffffff;
  --theme-color-text: #000000;
  --theme-color-highlight: #000000;
  --font-size: 100%;

  font-size: var(--font-size);
}

.layout {
  // margin-bottom compensates for the height of the fixed navbar
  padding-bottom: 6rem;
  background-color: $background-color;
  min-height: 100vh;
}
