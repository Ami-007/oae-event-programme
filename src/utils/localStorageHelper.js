export const getDarkTheme = () => {
  //   if (darkTheme == true) {
  //     localStorage.setItem("--background-color", darkTheme);
  //   } else {
  //     localStorage.setItem("theme", "light");
  //   }
  console.log(localStorage.getItem("darkMode"));

  // convert isDarkMode to a Boolean
  // return that boolean

  return true;
};

export const setDarkTheme = (isDarkMode) => {
  localStorage.setItem("darkMode", isDarkMode.toString());
};
