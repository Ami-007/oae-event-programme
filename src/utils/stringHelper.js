export const getFontType = (fontType) => {
  if (typeof fontType === "string") {
    fontType = parseInt(fontType);
  }
  if (fontType === 1) {
    return "modern";
  } else if (fontType === 2) {
    return "classic";
  } else {
    return "modern";
  }
};

export default getFontType;
