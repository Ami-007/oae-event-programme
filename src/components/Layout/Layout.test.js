import Layout from "./Layout";
import { customRender } from "../../utils/testUtils";

test("Renders the Layout component with default page", () => {
  const { container } = customRender(
    <Layout>
      <p>Sample children</p>
    </Layout>
  );

  expect(container).toMatchSnapshot();
});

test("Ensure the Layout div has a className", () => {
  const { container } = customRender(
    <Layout>
      {" "}
      <p>Sample children</p>{" "}
    </Layout>
  );
  expect(container.firstChild).toHaveClass("modern");
});
