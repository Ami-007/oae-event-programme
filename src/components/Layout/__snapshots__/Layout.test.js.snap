// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Renders the Layout component with default page 1`] = `
<div>
  <div
    class="modern"
  >
    <p>
      Sample children
    </p>
  </div>
</div>
`;
